src/main/java/com/javagirls/MangoRoom/service/ReservationService.java

```diff
@@ -94,4 +96,25 @@ public void changeReservationStatus(Long id, Status status) {
         return reservationRepository.findAll().stream()
                 .map((reservation) -> mapper.map(reservation, ReservationDto.class)).collect(Collectors.toList());
     }
+
+    public List<RoomDto> getFreeRooms(LocalDateTime startDate, LocalDateTime endDate){
+        List<ReservationDto> reserved = new ArrayList<>();
+        List<RoomDto> available = new ArrayList<>();
+
+        findAllReservationsDto().stream()
+                .filter(reservationDto -> (reservationDto.getCheckOut().truncatedTo(ChronoUnit.DAYS)
+                        .isAfter(startDate.truncatedTo(ChronoUnit.DAYS))
+                        && (reservationDto.getCheckIn().truncatedTo(ChronoUnit.DAYS)
+                        .isBefore(endDate.truncatedTo(ChronoUnit.DAYS)))))
+                .map(reservationDto -> reserved.add(reservationDto)).collect(Collectors.toList());
+
+        List<RoomDto> allRooms = roomService.findAllRooms();
+        for(RoomDto room : allRooms){
+            if(!reserved.contains(room.getReservations())){
```

Comment:
room.getReservations() zwraca rezerwacje dla danego pokoju, więc to nie ma prawa działać.
Musiałabyś przefiltrować listę wszystkich pokoi i odrzucić te, których id jest przypisane do rezerwacji z listy reserved.

Reply:
j.w.